File: algorithms/easy/largetst_positive_integer_that_exists_with_its_negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
      set<int> s;
      int ans = -1;
      for (auto n: nums) {
        s.insert(n);
        if (s.find(-n) != s.end()) {
          ans = max(ans, abs(n));
        }
      }
      return ans;
    }
};

File: algorithms/easy/largetst_positive_integer_that_exists_with_its_negative.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_max_k(nums: Vec<i32>) -> i32 {
        let mut ans = -1;
        let mut set = HashSet::new();

        for n in nums {
            set.insert(n);
            if set.contains(&-n) {
                ans = ans.max(n.abs());
            }
        }

        ans
    }
}
